Neo4j.Driver/Neo4j.Driver.Metrics/ConnectionPoolMetrics.cs

```diff
@@ -33,15 +33,15 @@ internal class ConnectionPoolMetrics : IConnectionPoolMetrics, IConnectionPoolLi
         private long _timedOutToAcquire;
 
         public int Creating => _creating;
-        public long Created => _created;
-        public long FailedToCreate => _failedToCreate;
+        public long Created => Interlocked.Read(ref _created);
```

Comment:
To me, I do not feel a need to read a very precise value for metrics. If the values in the test assertion are sometimes wrong, then maybe we shall just remove that assertion in the tests? 

I feel it might be a big performance impact if we always atomically read these values for metrics.

Reply:
This was a spike to see if there's some kind of concurrency issue there. I will continue working to hunt for a connection leakage - but I agree that we can revert this commit in all.

Reply:
Then why not also have Creating and others also to be read atomically?